visualization/include/pcl/visualization/impl/point_cloud_color_handlers.hpp

```diff
@@ -73,6 +73,47 @@ pcl::visualization::PointCloudColorHandlerCustom<PointT>::getColor (vtkSmartPoin
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointT> void
+pcl::visualization::PointCloudColorHandlerCustomIndividual<PointT>::setInputCloud (
+    const PointCloudConstPtr &cloud)
+{
+  PointCloudColorHandler<PointT>::setInputCloud (cloud);
+  
+  if (cloud_->points.size () == rgb_values_.size())
+    capable_ = true;
+  else
+    capable_ = false;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointT> bool
+pcl::visualization::PointCloudColorHandlerCustomIndividual<PointT>::getColor (vtkSmartPointer<vtkDataArray> &scalars) const
+{
+  if (!capable_ || !cloud_)
+    return (false);
+
+  if (!scalars)
```

Comment:
Does this check make sense? A vtkSmartPointer is a full blown object, not exactly an address. I think you need to get a hold of the pointer the smart pointer is holding with like [this](http://www.vtk.org/doc/nightly/html/classvtkSmartPointer.html#a864ded1a8d272f26b1f49c78f8348459) to make the check.